frontend/js/src/user/playlists/components/ImportPlaylistModal.tsx

```diff
@@ -0,0 +1,190 @@
+import React, { useCallback, useContext, useState } from "react";
+import NiceModal, { useModal } from "@ebay/nice-modal-react";
+import { toast } from "react-toastify";
+import GlobalAppContext from "../../../utils/GlobalAppContext";
+import { ToastMsg } from "../../../notifications/Notifications";
+import Loader from "../../../components/Loader";
+
+type ImportPLaylistModalProps = {
+  listenlist: JSPFTrack[];
+};
+
+export default NiceModal.create((props: ImportPLaylistModalProps) => {
+  const modal = useModal();
+
+  const closeModal = React.useCallback(() => {
+    modal.hide();
+    document?.body?.classList?.remove("modal-open");
+    setTimeout(modal.remove, 200);
+  }, [modal]);
+
+  const { APIService, currentUser } = React.useContext(GlobalAppContext);
+  const [playlists, setPlaylists] = React.useState<SpotifyPlaylistObject[]>([]);
+  const [loading, setLoading] = React.useState<string[]>([]);
+
+  React.useEffect(() => {
+    async function getUserPlaylistsFromSpotify() {
+      try {
+        const response = await APIService.importPlaylistToSpotify(
+          currentUser?.auth_token
+        );
+
+        if (!response) {
+          return;
+        }
+
+        setPlaylists(response);
+      } catch (error) {
+        toast.error(
+          <ToastMsg
+            title="Error loading playlists"
+            message={error?.message ?? error}
+          />,
+          { toastId: "load-playlists-error" }
+        );
+      }
+    }
+
+    if (currentUser?.auth_token) {
+      getUserPlaylistsFromSpotify();
+    }
+  }, [APIService, currentUser]);
+
+  const alertMustBeLoggedIn = () => {
+    toast.error(
+      <ToastMsg
+        title="Error"
+        message="You must be logged in for this operation"
+      />,
+      { toastId: "auth-error" }
+    );
+  };
+
+  const importTracksToPlaylist = async (
+    playlistID: string,
+    playlistName: string
+  ) => {
+    if (!currentUser?.auth_token) {
+      alertMustBeLoggedIn();
+      return;
+    }
+
+    if (loading.includes(playlistID)) {
+      setLoading(loading.filter((i: string) => i !== playlistID));
+    } else {
+      setLoading([...loading, playlistID]);
+    }
+
+    try {
+      const newPlaylist: JSPFPlaylist = await APIService.importSpotifyPlaylistTracks(
+        currentUser?.auth_token,
+        playlistID
+      );
+      toast.success(
+        <ToastMsg
+          title="Successfully imported playlist from Spotify"
+          message={
+            <>
+              Imported
+              <a href={newPlaylist.identifier}> {playlistName}</a>
+            </>
+          }
+        />,
+        { toastId: "create-playlist-success" }
+      );
+      setLoading(loading.filter((i: string) => i !== playlistID));
+      modal.resolve(newPlaylist);
```

Comment:
I guess this means we're closing the modal and updating the playlist page with that single playlist after the  import finishes, which is maybe not what you were planning on?

Not sure what to do about this then. We could store an array of successfully imported playlists and send that back once the user click on a "done" button for example, which would require a change in `onPlaylistCreated` to accept an array of JSPFPlaylist

Reply:
Oh man, I totally forgot about implementing that part. 

Im thinking, since this function of importing playlists are called by each playlist button separately, each of them can call the modal.resolve and pass the playlist the created (or this is not how it works, srry if im mistaken)

Reply:
I am pretty sure if you call .resolve it will resolve the promise that we created by using `NiceModal.show(...).then(...)`, and consequently only the first created playlist will be passed to `onPlaylistCreated`.

In order to resolve multiple playlists, let ;e expand on the solution I suggested above:
1. modify the resolved value function [here](https://github.com/metabrainz/listenbrainz-server/pull/2800/files#diff-50740f3f836f659fc01031697d7a52c784011746a687be5433890665a4d76cb1R181-R185) to accept a JSPFPlaylist or an array of JSPFPlaylist -> `NiceModal.show<JSPFPlaylist | JSPFPlaylist[], any>(...`
2. In the .then function callback just below, check if the `playlist` value is an array (`if(Array.isArray(...))`), and if so send each item in the array to `onPlaylistCreated` 
3.  Modify `ImportSpotifyPlaylistModal` to store successfully created playlists in an array in the component state
4. Create a new function `resolveAndClose` that will do two things: 1. call modal.resolve with the stored playlist array and 2. call the `closeModal` function
5. Make the close button of the modal call `resolveAndClose` instead of closeModal